docs/standard/serialization/binaryformatter-migration-guide/winforms-migration-guide.md

```diff
@@ -0,0 +1,89 @@
+- That property is not read-only.
+- That property does not have a DefaultValueAttribute.
+- That property does not have a respective `bool ShouldSerialize[PropertyName]` method that returns `false` at the time of the CodeDOM serialization process. (Note: the method can have `private` scope.)
+
+If these statements are true, the Designer determines if that property’s type has a type converter. If it does, the Designer uses the type converter to serialize the property content. Otherwise, it uses `BinaryFormatter` to serialize the content into the resource file.
+Windows Forms has added analyzers along with code fixes to help bring awareness to this type of behavior where `BinaryFormatter` serialization might be occurring without the developer’s knowledge.
+
+## Migrate away from BinaryFormatter
+
+If types that aren't intrinsically handled during serialization and deserialization are used in the affected scenarios, you'll need to take action to complete migration to .NET 9 or a later version.
+
+### Clipboard and drag-and-drop
+
+For types that aren't intrinsically handled that are used in clipboard and drag-and-drop operations, it's recommended that you format those types as a `byte[]` or `string` payload before passing the data to clipboard or drag-and-drop APIs. Using JSON is one way to achieve this. You'll need to make adjustments to handle receiving a JSON formatted type just as adjustments have been made to place JSON formatted types on clipboard or drag-and-drop operations. For more information on how to serialize and deserialize the type with JSON, see [How to write .NET objects as JSON (serialize)](../system-text-json/how-to.md).
+
+### Designer scenarios
+
+For types that aren't intrinsically handled during serialization into resources or code, such as in the case of the Designer with ResX and CodeDom serialization scenarios, the prescribed way of migrating away from `BinaryFormatter` is to ensure a `TypeConverter` is registered for the type or property that's participating in serialization. This way, during serialization and deserialization, the `TypeConverter` is used in lieu of where `BinaryFormatter` was once used. For more information on implementing a type converter, see [`TypeConverter` Class](/dotnet/api/system.componentmodel.typeconverter#notes-to-inheritors)
+
+## Compatibility workaround (not recommended)
+
+For users who cannot migrate away from `BinaryFormatter` for whatever reason, `BinaryFormatter` can be added back for compatibility. For more information, see [BinaryFormatter Migration Guide: Compatibility Package](compatibility-package.md).
+
+> [!CAUTION]
+> BinaryFormatter is dangerous and not recommended as it puts consuming apps at risk for attacks such as denial of service, which can render the app unresponsive and or unexpectedly terminate it. For more information about the risks `BinaryFormatter` poses, see [Deserialization risks in use of BinaryFormatter and related types](../binaryformatter-security-guide.md).
```

Comment:
I think the worst problem is that it runs untrusted code - Creates type whose name came from an unknown source, not the DoS